x-pack/legacy/plugins/siem/server/lib/note/saved_object.ts

```diff
@@ -104,21 +104,22 @@ export class Note {
     version: string | null,
     note: SavedNote
   ): Promise<ResponseNote> {
-    let timelineVersionSavedObject = null;
     try {
-      if (note.timelineId == null) {
-        const timelineResult = convertSavedObjectToSavedTimeline(
-          await this.libs.savedObjects
-            .getScopedSavedObjectsClient(request[internalFrameworkRequest])
-            .create(
-              timelineSavedObjectType,
-              pickSavedTimeline(null, {}, request[internalFrameworkRequest].auth || null)
-            )
-        );
-        note.timelineId = timelineResult.savedObjectId;
-        timelineVersionSavedObject = timelineResult.version;
-      }
       if (noteId == null) {
+        let timelineVersionSavedObject = null;
```

Comment:
optional: i have not tested the code below, but I the `let` can be a `const` with something like:

```
        const timelineVersionSavedObject = await (async () => {
          if (note.timelineId == null) {
            const timelineResult = convertSavedObjectToSavedTimeline(
              await this.libs.savedObjects
                .getScopedSavedObjectsClient(request[internalFrameworkRequest])
                .create(
                  timelineSavedObjectType,
                  pickSavedTimeline(null, {}, request[internalFrameworkRequest].auth || null)
                )
            );
            note.timelineId = timelineResult.savedObjectId;
            return timelineResult.version;
          }
        })();
```